pkg/policy/rule.go

```diff
@@ -191,5 +230,55 @@ func (r *rule) canReach(ctx *SearchContext, state *traceState) api.Decision {
 		}
 	}
 
+	return entitiesDecision
+}
+
+func (r *rule) canReachEntities(ctx *SearchContext, state *traceState) api.Decision {
+	entitiesFound := false
+	for _, r := range r.Egress {
+		for _, entity := range r.ToEntities {
+			entitiesFound = true
+			l, ok := api.EntitySelectorMapping[entity]
```

Comment:
Can we do this mapping at validation time and store it instead of doing this every time in the evaluation?

Reply:
We can, I will store proper selectors in `rule` type.